instrumentation/runtime-telemetry/runtime-telemetry-java8/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/java8/internal/ExperimentalFileDescriptor.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.runtimemetrics.java8.internal;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.metrics.Meter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * Registers measurements that generate experimental metrics about file descriptor.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public class ExperimentalFileDescriptor {
+  /** Register observers for java runtime experimental file descriptor metrics. */
+  public static List<AutoCloseable> registerObservers(OpenTelemetry openTelemetry) {
+    return registerObservers(
+        openTelemetry,
+        FileDescriptorMethods.openFileDescriptorCount(),
+        FileDescriptorMethods.maxFileDescriptorCount());
+  }
+
+  // Visible for testing
+  static List<AutoCloseable> registerObservers(
+      OpenTelemetry openTelemetry,
+      Supplier<Long> openFileDescriptorCount,
+      Supplier<Long> maxFileDescriptorCount) {
+    Meter meter = JmxRuntimeMetricsUtil.getMeter(openTelemetry);
+    List<AutoCloseable> observables = new ArrayList<>();
+
+    if (openFileDescriptorCount != null) {
+      observables.add(
+          meter
+              .gaugeBuilder("os.file.descriptor.open")
+              .setDescription("number of open file descriptors")
+              .setUnit("{file}")
+              .buildWithCallback(
+                  observableMeasurement -> {
+                    Long openCount = openFileDescriptorCount.get();
+                    if (openCount != null && openCount >= 0) {
+                      observableMeasurement.record(openCount);
+                    }
+                  }));
+    }
+
+    if (maxFileDescriptorCount != null) {
+      observables.add(
+          meter
+              .gaugeBuilder("os.file.descriptor.max")
```

Comment:
Host metrics receiver does not have this. It is similar to `jvm.memory.limit` and `jvm.buffer.memory.limit` that we already have. I'd use `limit` instead of `max` in the metric name. The descriptions of the existing limit metrics are `Measure of max obtainable memory.` and `Measure of total memory capacity of buffers.` The description of this metric should follow similar style.

Reply:
+1 on `.limit`

@xiangtianyu are these process metrics, or os-wide metrics? can you open an issue in https://github.com/open-telemetry/semantic-conventions to propose any new metrics, and add a comment in the code pointing to that issue?

Reply:
I've opened an issue https://github.com/open-telemetry/semantic-conventions/issues/1275